File: app/src/main/java/org/odddev/fantlab/core/storage/StorageModule.java
package org.odddev.fantlab.core.storage;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * @author kenrube
 * @since 29.09.16
 */

@Module
public class StorageModule {

    @Provides
    @Singleton
    StorageManager getStorageManager(Context context) {
        return new StorageManager(context);
    }
}

File: app/src/main/kotlin/org/odddev/fantlab/core/storage/StorageModule.kt
package org.odddev.fantlab.core.storage

import android.content.Context
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class StorageModule {

	@Provides
	@Singleton
	fun getStorageManager(context: Context) = StorageManager(context)
}
